lib/global-error/src/ext.rs

```diff
@@ -36,3 +30,78 @@ pub struct RetryError {
 	pub message: &'static str,
 	pub location: Location,
 }
+
+/// `UnwrapOrAssertError` is a trait used for handling unwrapping of `Result`
+/// and `Option` types.
+///
+/// This trait provides a method `assertion_error_unwrap` that takes a `Result`
+/// or `Option` and returns a `Result`. If the original `Result` or `Option` is
+/// `Ok` or `Some`, it returns `Ok` with the unwrapped value. If it's `Err` or
+/// `None`, it returns `Err` with an `AssertionError::Unwrap` that includes a
+/// message and location.
+pub trait UnwrapOrAssertError {
+	type WrappedType;
+
+	fn assertion_error_unwrap(
+		self,
+		message: String,
+	) -> Result<Self::WrappedType, crate::ext::AssertionError>;
+}
+
+impl<T, E: core::fmt::Debug> UnwrapOrAssertError for Result<T, E> {
+	type WrappedType = T;
+
+	fn assertion_error_unwrap(self, message: String) -> Result<T, crate::ext::AssertionError> {
+		match self {
+			Ok(t) => Ok(t),
+			Err(e) => Err(crate::ext::AssertionError::Unwrap {
+				message: format!("{}: {:?}", message, e),
+				location: crate::location!(),
```

Comment:
Was this ever updated to call `location!` in the macro instead of the function?

Reply:
I'm pretty sure I did swap it out, but I might have clobbered it somehow. Will reimplement.

Reply:
Got it sorted, thanks for seeing that again 👍